cmd/bg-prov/README.md

```diff
@@ -1 +1,333 @@
+        <password>              Password for AES256 encryption of private keys
+        [<path>]                Path to store keys. 
+                                File names are '<path>_bpm/.pub' and '<path>_km/.pub' respectivly
+```
+
+     
+```bash
+./bg-prov template                       Writes template JSON configuration into file
+        <path>                   Path to the newly generated JSON configuration file.
+
+        --revision            Platform Manufacturer’s BPM revision number.
+        --svn                 Boot Policy Manifest Security Version Number
+        --acmsvn              Authorized ACM Security Version Number
+        --nems                Size of data region need by IBB expressed in 4K pages. 
+                              E.g., value of 1 = 4096 bytes; 2 = 8092 bytes, etc. Must not be zero
+        --pbet                Protect BIOS Environment Timer (PBET) value.
+        --ibbflags            IBB Control flags
+        --mchbar              MCHBAR address
+        --vdtbar              VTDPVC0BAR address
+        --dmabase0            Low DMA protected range base
+        --dmasize0            Low DMA protected range limit
+        --dmabase1            High DMA protected range base.
+        --dmasize1            High DMA protected range limit.
+        --entrypoint          IBB (Startup BIOS) entry point
+        --sintmin             OEM authorized SinitMinSvn value
+        --txtflags            TXT Element control flags
+        --powerdowninterval   Duration of Power Down in 5 sec increments
+        --acpibaseoffset      ACPI IO offset.
+        --powermbaseoffset    ACPI MMIO offset.
+        --cmosoff0            CMOS byte in bank 0 to store platform wakeup time
+        --cmosoff1            Second CMOS byte in bank 0 to store platform wakeup time
+```
+
+Workflows
+==========
+
+I. Boot Policy / Key Manifest Generation/Signing/Stitching
+-------------------------------
+
+1. Create a template config file
+```bash
+./bg-prov template ./config.json
+```
+
+2. Create keys for signing of Key Manifest (KM) and Boot Policy Manifest (BPM)
+Algorithm: RSA, BitSize: 2048, no password for enryption of private key files
+```bash
+./bg-prov key-gen RSA2048 "" ./Keys/mykey
+```
+
+3. Generate Key Manifest (KM)
+```bash
+./bg-prov km-gen ./KM/km_unsigned.bin ./Keys/mykey_km.pub \
+        --config=./config.json \
+        --pkhashalg=12 \
+        --bpmpubkey=./Keys/mykey_bpm.pub \
+        --bpmhashalgo=12
+```
+
+4. Generation of Boot Policy Manifest (BPM)
+```bash
+./bg-prov bpm-gen ./BPM/bpm_unsigned.bin ./firmware.rom --config=./config.json
+```
+
+5. Sign Key Manifest (KM)
+```bash
+./bg-prov km-sign ./KM/km_unsigned.bin ./KM/km_signed.bin ./Keys/myKey_km.priv ""
+```
+
+6. Sign Boot Policy Manifest (BPM)
+```bash
+./bg-prov bpm-sign ./BPM/bpm_unsigned.bin ./BPM/bpm_signed.bin ./Keys/myKey_bpm.priv ""
+
+```
+
+7. Export ACM for stitching

```

Comment:
that only works if there's an ACM in the firmware image?

Reply:
otherwise you need to provide a ACM binary otherwise.